guides/token-factory/frontend/providers/walletConnectProvider.tsx

```diff
@@ -0,0 +1,235 @@
+"use client"
+import { createContext, ReactNode, useContext, useEffect, useState } from 'react';
+import { WalletConnectModal } from '@walletconnect/modal';
+import WCClient, {SignClient} from '@walletconnect/sign-client';
+import Logo from '@/assets/images/logo.png';
+import { SessionTypes } from '@walletconnect/types';
+import { getLisk32AddressFromPublicKey } from '@/utils/lisk';
+
+interface WalletConnectProps {
+	session: any
+	connect: () => void
+	disconnect: () => void
+	address: string | undefined
+	publicKey: string | undefined
+	sendTransaction: () => void
+}
+
+type TransactionResult = {
+	module?: string;
+	command?: string;
+	nonce?: string;
+	fee?: string;
+	senderPublicKey?: string;
+	params?: any;
+	signatures?: string[];
+	error?: { code: number; message: string };
+};
+
+export const WalletConnect = createContext<WalletConnectProps>(
+	{} as WalletConnectProps,
+);
+
+export const useWalletConnect = () => useContext(WalletConnect)
+
+export const WalletConnectProvider = ({ children }: {
+	children: ReactNode;
+}) => {
+	const [session, setSession] = useState<any>()
+	const [signClient, setSignClient] = useState<WCClient>();
+	const [topic, setTopic] = useState<string | undefined>();
+	const [address, setAddress] = useState<string | undefined>();
+	const [publicKey, setPublicKey] = useState<string | undefined>();
+
+	//const projectID = process.env.REACT_APP_WC_PROJECT_ID as string
+	const projectID = "730e84c79793076a98d0f8830ce1a1a9"
```

Comment:
Best to put this directly as an ENV especially because we work in a public repo.